st2common/tests/unit/services/test_packs.py

```diff
@@ -241,3 +257,156 @@ def test_exception_to_remove_resource_metadata_file(self, remove):
+
+    def test_clone_action_with_local_shell_cmd_runner(self):
+        clone_action(
+            TEST_SOURCE_PACK_PATH,
+            "actions/echo.yaml",
+            "",
+            "local-shell-cmd",
+            TEST_DEST_PACK_PATH,
+            TEST_DEST_PACK,
+            "action_3",
+        )
+        cloned_action_metadata_file_path = os.path.join(
+            TEST_DEST_PACK_PATH, "actions", "action_3.yaml"
+        )
+        self.assertTrue(os.path.exists(cloned_action_metadata_file_path))
+
+    def test_clone_workflow(self):
+        clone_action(
+            TEST_SOURCE_WORKFLOW_PACK_PATH,
+            "actions/data-flow.yaml",
+            "workflows/data-flow.yaml",
+            "orquesta",
+            TEST_DEST_PACK_PATH,
+            TEST_DEST_PACK,
+            "workflow_1",
+        )
+        cloned_workflow_metadata_file_path = os.path.join(
+            TEST_DEST_PACK_PATH, "actions", "workflow_1.yaml"
+        )
+        cloned_workflow_entry_point_file_path = os.path.join(
+            TEST_DEST_PACK_PATH, "actions", "workflows", "workflow_1.yaml"
+        )
+        self.assertTrue(os.path.exists(cloned_workflow_metadata_file_path))
+        self.assertTrue(os.path.exists(cloned_workflow_entry_point_file_path))
+
+    @mock.patch("shutil.copy")
+    def test_permission_error_to_write_in_destination_file(self, mock_copy):
+        mock_copy.side_effect = PermissionError("No permission to write in file")
+        cloned_action_entry_point_file_path = os.path.join(
+            TEST_DEST_PACK_PATH, "actions", "action_4.py"
+        )
+        expected_msg = 'No permission to write in "%s" file' % (
+            cloned_action_entry_point_file_path
+        )
+
+        with self.assertRaisesRegexp(PermissionError, expected_msg):
+            clone_action(
+                TEST_SOURCE_PACK_PATH,
+                "actions/inject_trigger.yaml",
+                "inject_trigger.py",
+                "python-script",
+                TEST_DEST_PACK_PATH,
+                TEST_DEST_PACK,
+                "action_4",
+            )
+
+    @mock.patch("shutil.copy")
+    def test_file_not_found_error_for_destination_file(self, mock_copy):
+        mock_copy.side_effect = FileNotFoundError("No such file or directory")
+        cloned_action_entry_point_file_path = os.path.join(
+            TEST_DEST_PACK_PATH, "actions", "action_5.py"
+        )
+        expected_msg = (
+            "Please make sure 'workflows' directory present in path: '%s'"
+            % (cloned_action_entry_point_file_path)
+        )
+
+        with self.assertRaisesRegexp(FileNotFoundError, expected_msg):
+            clone_action(
+                TEST_SOURCE_PACK_PATH,
+                "actions/inject_trigger.yaml",
+                "inject_trigger.py",
+                "python-script",
+                TEST_DEST_PACK_PATH,
+                TEST_DEST_PACK,
+                "action_5",
+            )
+
```

Comment:
- Add a unit test to check if destination pack does not exists.
- Add a unit test to check if destination files already exist.